src/Service.Tests/SqlTests/RestApiTests/Insert/InsertApiTestBase.cs

```diff
@@ -343,6 +343,94 @@ public virtual async Task InsertOneWithComputedFieldInRequestBody()
                 expectedSubStatusCode: DataApiBuilderException.SubStatusCodes.BadRequest.ToString()
                 );
         }
+
+        /// <summary>
+        /// Test to validate the successful creation of an item in the database through a
+        /// REST POST API request. This test also validates that the response returned takes
+        /// into account the fields configuration set for READ action of the role with which
+        /// the request was executed.
+        /// </summary>
+        [TestMethod]
+        public virtual async Task InsertOneWithExcludeFieldsTest()
+        {
+            string requestBody = @"
+            {
+                ""title"": ""My New Book"",
+                ""publisher_id"": 1234
+            }";
+
+            string expectedLocationHeader = $"id/{STARTING_ID_FOR_TEST_INSERTS}";
+            await SetupAndRunRestApiTest(
+                primaryKeyRoute: null,
+                queryString: null,
+                entityNameOrPath: _integrationEntityName,
+                sqlQuery: GetQuery(nameof(InsertOneWithExcludeFieldsTest)),
+                operationType: EntityActionOperation.Insert,
+                requestBody: requestBody,
+                expectedStatusCode: HttpStatusCode.Created,
+                expectedLocationHeader: expectedLocationHeader,
+                clientRoleHeader: "policy_tester_excludefields"
+            );
+        }
+
+        /// <summary>
+        /// Test to validate the successful creation of an item in the database through a
+        /// REST POST API request. This test also validates that an empty response is returned
+        /// when no read action is configured for the role with which the POST request is executed.
+        /// </summary>
+        /// <returns></returns>        
+        [TestMethod]
+        public virtual async Task InsertOneWithNoReadPermissionsTest()
```

Comment:
This is just a special case of the test above. Can club the two into one with appropriate comments.

Reply:
Discussed this offline: 

Adding separate methods makes the debugging easier and the test report generated in the Azure pipelines is more representative of all the test case results.

Let's say 4 test cases are combined in the same method. If suppose the 2nd test case fails, the test report only displays the method name. It does not specify which test case among the 4 failed. Also, if the 2nd test case fails, the 3 and 4th (rest of the test cases after it) do not run. So, potentially there could be more failures and more tests needs to be looked at. Also, care should be taken so that none of these 4 test cases interfere with each other in any aspect as the database is not reset after each test case execution.


The only downside to writing different methods for each test case could be that there is an additional overhead involved in terms of resetting the database and slightly longer time for the pipeline run. 
This could be a bigger discussion topic/refactor on how we choose to write these integration tests. 

At least, at the moment, it seems like it should be fine writing individual methods for different test cases, so leaving all the tests as is.

Reply:
Unclear result data would only occur if each test was copy/pasted in sequence verbatim into a single test. If you used `[DataRow(<args>, DisplayName="TestName")]` then you could just provide params for the fields you're changing. It seems like you're only changing 3-4 args.

Reply:
I considered using `DataTestMethod` and adding each of the test cases as a `DataRow` but that would not allow us to add relevant details about the test which will be helpful during debugging in case of test failures - Details such as what is the expected response, reasons for the expected response (lack of read permissions, presence of a database policy etc.).